src/irc/message.rs

```diff
@@ -4,27 +4,35 @@ pub struct Message<'a> {
   pub params: Vec<&'a str>,
 }
 
-impl Message<'_> {
-  pub fn from_string<'m>(line: &'m str) -> Result<Message<'m>, std::io::Error> {
+impl<'m> Message<'m> {
+  pub fn from_string(line: &'m str) -> Option<Message<'m>> {
     let mut parts = line.split(' ');
-    let prefix = if line.get(0..1).unwrap() == ":" {
-      parts.next()
-    } else {
-      None
-    };
 
-    Ok(Message {
+    let prefix = line.chars().nth(0)
+        .and_then(|c| {
```

Comment:
I like this!  I didn't know about [and_then](https://doc.rust-lang.org/rust-by-example/error/option_unwrap/and_then.html)

Reply:
Yeah, sometimes the combinators work out nicely (but sometimes they are a bit awkward, it depends on the specific use case)